LibVLCSharp/Shared/Core.cs

```diff
@@ -64,73 +67,114 @@ static void InitializeAndroid()
-                if (appExecutionDirectory == null)
-                    throw new NullReferenceException(nameof(appExecutionDirectory));
+                // Initializes X threads before calling VLC. This is required for vlc plugins like the VDPAU hardware acceleration plugin.
+                if (Native.XInitThreads() == 0)
+                {
+#if !NETSTANDARD1_1
+                    Trace.WriteLine("XInitThreads failed");
 #endif
+                }
             }
 
-            if (PlatformHelper.IsWindows)
+            // full path to directory location of libvlc and libvlccore has been provided
+            if (!string.IsNullOrEmpty(appExecutionDirectory))
             {
-                var arch = PlatformHelper.IsX64BitProcess ? ArchitectureNames.Win64 : ArchitectureNames.Win86;
+                Load(appExecutionDirectory);
+                return;
+            }
 
-                var librariesFolder = Path.Combine(appExecutionDirectory, Constants.LibrariesRepositoryFolderName, arch);
+            var paths = ComputeLibVLCSearchPaths();
 
-                _libvlccoreHandle = PreloadNativeLibrary(librariesFolder, $"{Constants.CoreLibraryName}.dll");
-                
-                if(_libvlccoreHandle == IntPtr.Zero)
-                {
-                    throw new VLCException($"Failed to load required native library {Constants.CoreLibraryName}.dll");
-                }
+            foreach(var path in paths)
+            {
+                if (Load(path, shouldThrow: false)) break;
+            }
 
-                _libvlcHandle = PreloadNativeLibrary(librariesFolder, $"{Constants.LibraryName}.dll");
-                
-                if(_libvlcHandle == IntPtr.Zero)
-                {
-                    throw new VLCException($"Failed to load required native library {Constants.LibraryName}.dll");
-                }
+            if(!Loaded)
+            {
+                throw new VLCException($"Failed to load required native libraries");
             }
-            else if (PlatformHelper.IsMac)
+        }
+
+        static List<string> ComputeLibVLCSearchPaths()
+        {
+            var paths = new List<string>();
+
+#if !NETSTANDARD1_1
+            // compute paths
+            var appExecutionDirectory = Path.GetDirectoryName(typeof(LibVLC).Assembly.Location);
+            var arch = PlatformHelper.IsX64BitProcess ? ArchitectureNames.Win64 : ArchitectureNames.Win86;
+            var librariesFolder = Path.Combine(appExecutionDirectory, Constants.LibrariesRepositoryFolderName, arch);
+            paths.Add(librariesFolder);
+#endif
+            // TODO: add more path
+            return paths;
+        }
+
+        static bool Load(string appExecutionDirectory, bool shouldThrow = true, bool log = true)
+        {
+            if (string.IsNullOrEmpty(appExecutionDirectory)) return false;
+
+            Debug.Assert(PlatformHelper.IsWindows || PlatformHelper.IsMac); // for now
```

Comment:
What does this do on linux? stop execution? If so, I'd prefer a `if(PlatfotmHelper.IsLinux) return;` or something less critical for the app.

Reply:
I'd prefer to be able to load the library on linux of course, I have a library I built and can certainly help with that code.